app/airq/air_quality.py

```diff
@@ -23,9 +23,10 @@
 MAX_NUM_NEARBY_ZIPCODES = 200
 
 
-@dataclasses.dataclass
+@dataclasses.dataclass(frozen=True)
```

Comment:
This makes it so that instances of this class can't be changed after they've been created. We probably want this since there's no reason to be writing metrics data for a zipcode more than once per request.